packages/colorpicker/src/elements/ColorDialog/index.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState, useRef, forwardRef, HTMLAttributes } from 'react';
+import PropTypes from 'prop-types';
+import mergeRefs from 'react-merge-refs';
+import { GARDEN_PLACEMENT } from '@zendeskgarden/react-modals';
+import Chevron from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';
+import { ColorPicker, IColorPickerProps } from '../ColorPicker';
+import { IRGBColor, IColorPickerState } from '../ColorPicker/reducer';
+import {
+  StyledIcon,
+  StyledButton,
+  StyledFauxInput,
+  StyledDialogPreview,
+  StyledTooltipModal,
+  StyledTooltipBody
+} from '../../styled';
+
+export interface IColorDialogProps extends IColorPickerProps {
+  /** A callback function that is called when the color dialog is closed */
+  onClose?: (selectedColor: string | IRGBColor | IColorPickerState) => void;
```

Comment:
Please re-phrase per specified grammar RFC. All functions must be fully JSDoc'd. Consult the `website` for info.